### PR TITLE
Docker build engine on Windows

### PR DESCRIPTION
**Pull request by vigejolla**
Note: Only the last commit belongs to this PR

### CODE REVIEW

File: src/libs/sfdk/dockervirtualmachine.cpp [hunk 1]
```diff
@@ -83,12 +84,20 @@ QString dockerPath()
     if (!path.isNull())
         return path;
 
+    QString dockerBinary = DOCKER;
+    QString pathVariable = "PATH";
+    QChar pathSeparator = ':';
```
Comment (martyone) [inline src/libs/sfdk/dockervirtualmachine.cpp]: QDir::listSeparator()

File: src/libs/sfdk/dockervirtualmachine.cpp [hunk 2]
```diff
@@ -83,12 +84,20 @@ QString dockerPath()
...
+    if (Utils::HostOsInfo::isWindowsHost()) {
+        dockerBinary.append(".exe");
```
Comment (martyone) [inline src/libs/sfdk/dockervirtualmachine.cpp]: QTC_HOST_EXE_SUFFIX

File: src/libs/sfdk/dockervirtualmachine.cpp [hunk 3]
```diff
@@ -83,12 +84,20 @@ QString dockerPath()
...
+        pathVariable = "Path";
```
Comment (martyone) [inline src/libs/sfdk/dockervirtualmachine.cpp]: I think this is not needed. The docs for QProcessEnvironment says "On Windows, the variable names are case-insensitive, but case-preserving. QProcessEnvironment behaves accordingly." and looging at QProcessEnvKey in qproess_p.h it really seems to be so.

**Comment by martyone** [comment]
Now I noticed `Utils::Environment::systemEnvironment().searchInPath(executable)` is used elsewhere that does all the necessary tricks.

**Comment by vigejolla** [comment]
> Now I noticed `Utils::Environment::systemEnvironment().searchInPath(executable)` is used elsewhere that does all the necessary tricks.

nice

File: src/libs/sfdk/dockervirtualmachine.cpp [hunk 4]
```diff
@@ -28,7 +28,9 @@
 #include "sfdkconstants.h"
 #include "usersettings_p.h"
 
+#include <utils/environment.h>
 #include <utils/fileutils.h>
+#include <utils/hostosinfo.h>
```
Comment (martyone) [inline src/libs/sfdk/dockervirtualmachine.cpp]: Is this still needed?

Comment (vigejolla) [reply src/libs/sfdk/dockervirtualmachine.cpp]: nope. removed.

**Pull request merged**